src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -14,24 +14,28 @@ public class CommandResult {
     /** Help information should be shown to the user. */
     private final boolean showHelp;
 
+    /** The application should show user profile. */
+    private final boolean showUserProfile;
+
     /** The application should exit. */
     private final boolean exit;
 
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
-    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit, boolean showUserProfile) {
```

Comment:
LGTM. Just in the future if you're adding onto this, perhaps use polymorphism for the constructor? I see that most of the constructor invocations have this field as false; maybe have another constructor with just CommandResult(feedback, showHelp, exit) with showUserProfile by default false? Three booleans in a row is fine, but if it is going to continue expanding, it becomes quite hard to maintain many commands at once.

Just something to consider if want to add more stuff to it. But if it's going to remain as this method signature, then it's fine.

Reply:
Yes, agreed. Thanks :)